### ISSUE TITLE
Change Request: Simplify Bun detection in tests

### ISSUE THREAD

**Issue by sethamus** [eslint/markdown#540]
### Environment

ESLint version:
@eslint/markdown version:
Node version:
npm version:
Operating System:

### What problem do you want to solve?

Currently, in several tests, we skip running code on Bun using a Node-specific check like this:

```js
...(typeof process !== "undefined" &&
    process.release?.name === "node" &&
    !process.versions?.bun
        ? [
              {
                  code: `
                  [Grüsse]: https://example.com/
                  [Grüsse]: https://example.com/
                  `,
                  options: [{ allowDefinitions: ["GRÜẞE"] }],
              },
              {
                  code: `
                  [^Grüsse]: Grüsse
                  [^Grüsse]: Grüsse
                  `,
                  options: [{ allowFootnoteDefinitions: ["GRÜẞE"] }],
              },
          ]
        : []),
```

### What do you think is the correct solution?

According to the [Bun documentation](https://bun.com/guides/util/detect-bun), this can be simplified to:

```js
if (typeof Bun === "undefined") {
  // this code will only run when NOT using Bun
}

// OR

if (process.versions.bun) {
  // this code will only run when the file is run with Bun
}
```

### Participation

- [x] I am willing to submit a pull request for this change.
